test/deprecations.js

```diff
@@ -0,0 +1,50 @@
+var morgan = require('..')
+var assert = require('assert')
+
+describe('morgan() deprecations', function () {
+  var originalConsoleError
+  var errors = []
+  function getErrors () {
+    return errors.map(function (e) { return removeANSI(e.toString('utf8')) })
+  }
+  before(function () {
+    originalConsoleError = process.stderr.write
```

Comment:
Personally, I don't bother writing tests for these deprecation messages at all, but if you feel so inclined, I would say please don't do it by replacing `process.stderr.write` -- mocha itself may try to write there. Instead capture the messages through a listener: https://www.npmjs.com/package/depd#processondeprecation-fn

Reply:
I took with from https://github.com/dougwilson/nodejs-depd/blob/master/test/support/capture-stderr.js

Maybe you should also fix that :)

Reply:
No, because that is the module itself, to validate its writing to stderr. Modules embedding depd should use the listener hook only (i.e. they should follow the README).

Reply:
Ok, Please notice that your personal decision of not having tests for deprecations is what actually made it possible to have the issue that I reported initially. If you don't test it then errors on those message will get through.

You are too picky dude. It is ok, nothing personal, but I don't have more time to spend on this, that's why I closed it. I just wanted to help but I think that we are both losing time here. If you do it yourself then that would be faster.

Good luck there !